vmcnet/examples/hubbard_model.py

```diff
@@ -0,0 +1,66 @@
+from typing import Callable, Union
```

Comment:
Not too sure we need this file, actually. The HubbardAnsatz class seems to be more-or-less a slightly more restricted version of your `SimpleSlater` model? And I think the `make_hubbard_local_energy` can probably be moved to the test file where it is used atm.